### PR TITLE
refactor: introduce containerImageScanStatusPatch to reduce code duplication

### PR DESCRIPTION
**Pull request by erikgb**
This eliminates some delicate duplicate code and makes the code more DRY.

### CODE REVIEW

File: internal/controller/stas/containerimagescan_status.go [hunk 1]
```diff
@@ -28,8 +36,80 @@ func newContainerImageStatusPatch(cis *stasv1alpha1.ContainerImageScan) *stasv1a
...
+}
+
+type containerImageScanStatusPatch struct {
+	cis             *stasv1alpha1.ContainerImageScan
+	patch           *stasv1alpha1ac.ContainerImageScanApplyConfiguration
+	vulnerabilities []stasv1alpha1.Vulnerability
+	minSeverity     *stasv1alpha1.Severity
+}
+
+func (b *containerImageScanStatusPatch) withCondition(c *metav1ac.ConditionApplyConfiguration) *containerImageScanStatusPatch {
```
Comment (tenstad) [inline internal/controller/stas/containerimagescan_status.go:53]: Why `b` as receiver name? :D

File: internal/controller/stas/containerimagescan_status.go [hunk 2]
```diff
@@ -28,8 +36,80 @@ func newContainerImageStatusPatch(cis *stasv1alpha1.ContainerImageScan) *stasv1a
...
+
+func (b *containerImageScanStatusPatch) fromScanJob(job *batchv1.Job) *containerImageScanStatusPatch {
+	b.patch.Status.
+		WithLastScanTime(metav1.Now()).
+		WithLastScanJobUID(job.UID)
+
+	return b
+}
+
+func (b *containerImageScanStatusPatch) fromCompletedScanJob(job *batchv1.Job, vulnerabilities []stasv1alpha1.Vulnerability, minSeverity stasv1alpha1.Severity) *containerImageScanStatusPatch {
```
Comment (tenstad) [inline internal/controller/stas/containerimagescan_status.go:67]: Not sure about the `fromX` names. Based on the name alone, I would think it creates a new patch from the _from_ thing alone, not updating an existing one. `withCompleteScanJob` or something would make more sense to me.

File: internal/controller/stas/containerimagescan_status.go (hunk 1 above)
Comment (erikgb) [reply internal/controller/stas/containerimagescan_status.go]: From the previous name, ....Builder. Fixed now.

File: internal/controller/stas/containerimagescan_status.go [hunk 3]
```diff
@@ -28,8 +36,80 @@ func newContainerImageStatusPatch(cis *stasv1alpha1.ContainerImageScan) *stasv1a
...
+	p.patch.Status.
+		WithVulnerabilitySummary(vulnerabilitySummary(vulnerabilities, minSeverity)).
+		WithLastScanTime(now).
+		WithLastScanJobUID(job.UID).
+		WithLastSuccessfulScanTime(now)
+
+	return p
+}
+
+func (p *containerImageScanStatusPatch) execute(ctx context.Context, c client.Client) error {
```
Comment (tenstad) [inline internal/controller/stas/containerimagescan_status.go:81]: Is `apply` more fitting for a _patch_?

File: internal/controller/stas/containerimagescan_status.go (hunk 2 above)
Comment (erikgb) [reply internal/controller/stas/containerimagescan_status.go]: Agreed, and it used to be like that. Updated now.

File: internal/controller/stas/containerimagescan_status.go (hunk 3 above)
Comment (erikgb) [reply internal/controller/stas/containerimagescan_status.go]: Thanks! Much better. Changed name.

**Review by tenstad — APPROVED** [review]

**Pull request merged**